### DISCUSSION TITLE
Ubuntu: Install unstable packages

### DISCUSSION THREAD

**Discussion by idleberg** [community/community#25120] [Actions]
<p>I would like to install unstable packages in my GitHub action. Currently, I’m doing the same on Circle CI, but I would love to migrate to use GitHub Actions.</p>
<pre><code>name: MakeNSIS

on: [push]

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - name: Install NSIS
      run: |
        sudo bash -c 'echo "deb http://ftp.us.debian.org/debian unstable main contrib non-free" &gt;&gt; /etc/apt/sources.list.d/unstable.list'
        sudo apt-get -y update
        sudo apt-get -y install -t unstable nsis
    - name: Print NSIS version
      run: makensis -VERSION
</code></pre>
<p>This will fail due to a signature verification error:</p>
<pre><code>Reading package lists...
W: GPG error: http://ftp.us.debian.org/debian unstable InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 04EE7237B7D453EC NO_PUBKEY 648ACFD622F3D138
E: The repository 'http://ftp.us.debian.org/debian unstable InRelease' is not signed.
##[error]Process completed with exit code 100.
</code></pre>
<p>However, everything works fine when I’m trying to install the stable package instead:</p>
<pre><code>- name: Install NSIS
      run: |
        sudo apt-get -y update
        sudo apt-get -y install nsis
</code></pre>
<p>Any ideas how to fix this? As I’ve mentioned before, the same workflow works fine on Circle CI!</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The immediate reason for the problem is that you’re trying to install a Debian package (signed with a Debian key) on Ubuntu. Theoretically you could add the Debian signing keys to the apt keyring on the runner (see the debian-archive-keyring and debian-keyring packages), but I’d advise against that, because mixing distributions can cause instability even if they’re closely related.</p>
<p>You could run just run your whole job in a Debian unstable container, like so:</p>
<pre><code>jobs:
  build:
    runs-on: ubuntu-latest
    container: debian:sid
    steps:
      - ...
</code></pre>
<p>That way you have a full Debian unstable system, with all signing keys and packages available.</p>

**Comment by idleberg** [comment]
<p>Thank you, that works well. I also had to omit the <code>sudo</code>.</p>
